### ISSUE TITLE
How did you do action detection on ActEV dataset?and how did you overcome class imbalance in ActEV VIRAT?

### ISSUE THREAD

**Issue by malarsaravanan1991** [google/next-prediction#16]
Hey hi,

Interesting work. I am trying to compare my work (action detection)with your paper. So in the paper appendix you had mentioned the action detection experiment and the mAP is 0.155 if you use the detected labels as the predicted one. May i know the path to the code that does action detection in your repository ? and i saw that ActEV virat dataset has heavy class imbalance ,so in that case can you please let me know what you did you do to overcome that to train your network?

I am using BCELosswithlogits i see that you had used the same loss function in the tensorflow. The loss is not performing well for this dataset. 
So if possible let me know how you did action detection in your code and also any additional hacks you had applied to overcome class imbalance?

**Comment by JunweiLiang** [comment]
For the activity detection experiment (train an activity detection model and use the activity output as prediction), you only need to change the preprocess for training and validation set (no need to change the model code). Specifically, replace the future activity list with the observed activity list [here](https://github.com/google/next-prediction/blob/master/code/preprocess.py#L565). This way you will be training the same model for activity detection rather than prediction. The test set should remain the same for that experiment. 
We did not specifically address the class imbalance issue and did not try any bells and whistles. Hence the current low mAP (0.192).

**Comment by malarsaravanan1991** [comment]
Thanks

**Comment by malarsaravanan1991** [comment]
Quick question on mAP for ActDetect on ActEV VIRAT dataset. The reported mAP of 0.155 is on the test set or validation set? and is it frame level mAP or video level mAP?
